numpy/polynomial/polyutils.py

```diff
@@ -542,17 +542,15 @@ def _valnd(val_f, c, *args):
     c, args :
         See the ``<type>val<n>d`` functions for more detail
     """
-    try:
-        args = tuple(np.array(args, copy=False))
-    except Exception:
-        # preserve the old error message
-        if len(args) == 2:
+    args = [np.asanyarray(a) for a in args]
+    shape0 = args[0].shape
+    if not all((a.shape == shape0 for a in args)):
+        if len(args) ==3:
             raise ValueError('x, y, z are incompatible')
-        elif len(args) == 3:
+        elif len(args) ==2:
```

Comment:
```suggestion
        elif len(args) == 2:
```
Wow, that was pretty dumb of me, good catch